refactor: process conversation rename event - WPB-10177

johnxnguyen opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10177" title="WPB-10177" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-10177</a>  [iOS] Process ConversationRenameEvent
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove the jira markers to link tickets automatically -->


### Issue

Implement `ConversationRenameUpdateEvent`.

### Testing

_Describe how to test._

_Optional: attachments like images, videos, etc._

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

vytis commented on Wire-iOS/Sources/UserInterface/Conversation/Create/SimpleTextFieldValidator.swift:
```diff
@@ -34,13 +35,28 @@ final class SimpleTextFieldValidator: NSObject {
+        catch let stringValidationError as NSError {
+            
+            switch stringValidationError.code {
+            case Int(ZMManagedObjectValidationErrorCode.objectValidationErrorCodeStringTooLong.rawValue):
```
I wonder if there is a way to Swiftify this... Seems a bit clunky to use and also it replaces `validatedString` with a sanitized value that we don't use outside of CoreData.

mikeger replied:
I think that is fine, since the `SimpleTextFieldValidator` is wrapping the base validator (`StringLengthValidator`) that is used form ObjC into nice swift API.

johnxnguyen committed:
fix proteus message event payload

# Conflicts:
#	WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/Conversation/ConversationProteusMessageAddEventDecoder.swift

johnxnguyen committed:
use a proxy to decode events

johnxnguyen committed:
add Codable conformance to UpdateEvent

johnxnguyen committed:
move StoredUpdateEvent definition to data model

johnxnguyen committed:
add eventData attribute to StoredUpdateEvent

johnxnguyen committed:
introduce UpdateEventsRepository

johnxnguyen committed:
fix proteus error raw values

johnxnguyen committed:
test update event decryptor

johnxnguyen committed:
add proteus message decryptor tests

johnxnguyen committed:
Merge branch 'develop' into refactor/create-update-events-repository-wpb-8742

johnxnguyen committed:
store event envelopes instead

johnxnguyen committed:
Merge branch 'refactor/create-update-events-repository-wpb-8742' of github.com:wireapp/wire-ios into refactor/create-update-events-repository-wpb-8742

johnxnguyen committed:
Merge branch 'develop' into refactor/create-update-events-repository-wpb-8742

johnxnguyen committed:
restore StoredUpdateEvent changes

johnxnguyen committed:
lint

johnxnguyen committed:
fix tests

johnxnguyen committed:
lint

johnxnguyen committed:
rename method

johnxnguyen committed:
remove todo

johnxnguyen committed:
move to new file

johnxnguyen committed:
remove todo

johnxnguyen committed:
throw error when no last event id

johnxnguyen committed:
validate external data too

johnxnguyen committed:
Merge branch 'develop' into refactor/create-update-events-repository-wpb-8742

johnxnguyen committed:
lint

johnxnguyen committed:
Merge branch 'refactor/create-update-events-repository-wpb-8742' of github.com:wireapp/wire-ios into refactor/create-update-events-repository-wpb-8742

johnxnguyen committed:
improve naming

johnxnguyen committed:
remove stored update envelope optionality

johnxnguyen committed:
refactor: fetch stored events from the database - WPB-9907

johnxnguyen committed:
lint

johnxnguyen committed:
Merge branch 'develop' into refactor/create-update-events-repository-wpb-8742

# Conflicts:
#	WireDomain/WireDomain.xcodeproj/project.pbxproj

johnxnguyen committed:
Merge branch 'refactor/create-update-events-repository-wpb-8742' into refactor/fetch-events-WPB-9907

johnxnguyen committed:
wip: sync manager

johnxnguyen committed:
test live to suspended state

johnxnguyen committed:
cancel quick sync when suspending

johnxnguyen committed:
add test stub

johnxnguyen committed:
remove import

johnxnguyen committed:
Merge branch 'develop' into refactor/create-update-events-repository-wpb-8742

# Conflicts:
#	WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/UpdateEventDecodingProxyError.swift

johnxnguyen committed:
Merge branch 'refactor/create-update-events-repository-wpb-8742' into refactor/fetch-events-WPB-9907

johnxnguyen committed:
Merge branch 'refactor/fetch-events-WPB-9907' into refactor/create-sync-manager-wpb-9923

johnxnguyen committed:
add suspension tests

johnxnguyen committed:
add re-entrant todos

johnxnguyen committed:
remove slow sync

johnxnguyen committed:
groom

johnxnguyen committed:
add todo

johnxnguyen committed:
groom

johnxnguyen committed:
groom

johnxnguyen committed:
groom

johnxnguyen committed:
Merge branch 'develop' into refactor/create-sync-manager-wpb-9923

# Conflicts:
#	WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/Event decoding/UpdateEventDecodingProxy.swift
#	WireAPI/Sources/WireAPI/Models/Messaging/MessageContent.swift
#	WireAPI/Sources/WireAPI/Models/UpdateEvent/UpdateEventEnvelope/UpdateEventEnvelope.swift
#	WireDomain/Sources/WireDomain/Repositories/UpdateEventsRepository.swift
#	WireDomain/Sources/WireDomain/Repositories/UpdateEventsRepositoryError.swift
#	WireDomain/Sources/WireDomain/Synchronization/SyncState.swift
#	WireDomain/Sources/WireDomainSupport/Sourcery/generated/AutoMockable.generated.swift
#	WireDomain/Tests/WireDomain/Repositories/UpdateEventsRepositoryTests.swift
#	WireDomain/WireDomain.xcodeproj/project.pbxproj

johnxnguyen committed:
wip: quick sync with async stream

johnxnguyen committed:
groom

johnxnguyen committed:
add logs

johnxnguyen committed:
remove todos

johnxnguyen committed:
store last event id after processing live event

johnxnguyen committed:
add logs

johnxnguyen committed:
move extension to new file

johnxnguyen committed:
Merge branch 'develop' into refactor/create-sync-manager-wpb-9923

# Conflicts:
#	WireDomain/Sources/WireDomain/Repositories/UpdateEventsRepository.swift
#	WireDomain/WireDomain.xcodeproj/project.pbxproj

johnxnguyen committed:
fix tests

johnxnguyen committed:
delete dead code

johnxnguyen committed:
remove combine import

johnxnguyen committed:
remove dead code

johnxnguyen committed:
lint

johnxnguyen committed:
lint

johnxnguyen committed:
refactor: process conversation rename event - WPB-10177
